is/jsonable_test.ts

```diff
@@ -0,0 +1,163 @@
+import { assertEquals } from "@std/assert";
+import { isJsonable } from "./jsonable.ts";
+import { buildTestcases } from "./custom_jsonable_test.ts";
+
+Deno.test("isJsonable", async (t) => {
+  for (const [name, value] of buildTestcases()) {
+    switch (name) {
+      case "undefined":
+      case "bigint":
+      case "function":
+      case "symbol":
+        await t.step(`return false for ${name}`, () => {
+          assertEquals(isJsonable(value), false);
+        });
+        break;
+      default:
+        await t.step(`return true for ${name}`, () => {
+          assertEquals(isJsonable(value), true);
+        });
+    }
+  }
+
+  for (const [name, value] of buildTestcases()) {
+    switch (name) {
+      case "undefined":
+      case "null":
+        // Skip undefined, null that is not supported by Object.assign.
+        continue;
+      case "bigint":
+      case "function":
+        // Object.assign() doesn't make bigint, function Jsonable.
+        await t.step(
+          `return false for ${name} even if it is wrapped by Object.assign()`,
+          () => {
+            assertEquals(
+              isJsonable(
+                Object.assign(value as NonNullable<unknown>, { a: 0 }),
+              ),
+              false,
+            );
+          },
+        );
+        break;
+      default:
+        // Object.assign() makes other values Jsonable.
+        await t.step(
+          `return true for ${name} if it is wrapped by Object.assign()`,
+          () => {
+            assertEquals(
+              isJsonable(
+                Object.assign(value as NonNullable<unknown>, { a: 0 }),
+              ),
+              true,
+            );
+          },
+        );
+    }
+  }
+
+  for (const [name, value] of buildTestcases()) {
+    switch (name) {
+      case "undefined":
+      case "null":
+        // Skip undefined, null that is not supported by Object.assign.
+        continue;
+      case "bigint":
+      case "function":
+        // toJSON method assigned with Object.assign() makes bigint, function Jsonable.
+        await t.step(
+          `return true for ${name} if it has own toJSON method`,
+          () => {
+            assertEquals(
+              isJsonable(
+                Object.assign(value as NonNullable<unknown>, {
+                  toJSON: () => "custom",
+                }),
+              ),
+              true,
+            );
+          },
+        );
+        break;
```

Comment:
Is this block unnecessary?
Or would it be okay to fall through to the `default`?

Reply:
It's OK but for clarify.

Reply:
I do an "eyeball diff" to see if there are any differences in the test code between the `case ...` block and the `default` block.
I prefer code with `case ...`, comments and fall-through like:
```suggestion
      case "bigint":
      case "function":
        // toJSON method assigned with Object.assign() makes bigint, function Jsonable.
```

Reply:
Good point. I'll take it.